### PR TITLE
refactor: logging, manually mode, args, search logic

### PR DESCRIPTION
**Pull request by Nemental**
- changed shebang more flexible
- switched to `argparse` for command-line argument parsing, allowing users to specify the installation directory and whether to prompt for adoption confirmation
- added a `print_colored` function for more consistent colored logging
- enhanced `check_cask_available` to include searching for app name variations and providing alternatives when the exact cask is not found
- added `brew_search` for a better search for available brew applications
- added `choose_alternative_cask` to prompt users for selecting alternative cask names
- added `prompt_yes_no` for clear yes/no user prompts (needed for manually mode)
- simplified `brew_install` by removing redundant checks
- enhanced `brew_install` to force install if version differs
- updated and flattened main function logic to handle the adoption of applications more robustly and clearly
- improved error handling and messaging throughout the script

### CODE REVIEW

File: brew_cask_and_adopt_manual_installed_applications/brew_cask_and_adopt_manual_installed_applications.py [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
+import argparse
 
 
 # ANSI color codes for terminal output
-class colors:
-    """
-    Declare color variables
-    """
-
+class Colors:
```
Comment (xFuture603) [inline brew_cask_and_adopt_manual_installed_applications/brew_cask_and_adopt_manual_installed_applications.py:15]: Could you re-add the docstring for this class?

File: brew_cask_and_adopt_manual_installed_applications/brew_cask_and_adopt_manual_installed_applications.py [hunk 2]
```diff
@@ -127,85 +195,107 @@ def is_default_apple_app(app_name):
...
-        print(f"\n{colors.RED}Applications not found as Homebrew casks:{colors.ENDC}\n")
-        for app in not_found_apps:
+    if non_brew_managed_apps:
+        print_colored("\nApplications not found as Homebrew casks:", Colors.RED)
+        for app in non_brew_managed_apps:
             print(app)
-        print(
-            "\nFor more available Homebrew Casks, visit https://formulae.brew.sh/cask/"
-        )
+        print("For more available Homebrew Casks, visit https://formulae.brew.sh/cask/")
```
Comment (xFuture603) [inline brew_cask_and_adopt_manual_installed_applications/brew_cask_and_adopt_manual_installed_applications.py:279]: Could you add an empty line before this print statement? Currently, it looks a bit too cluttered IMO.

File: brew_cask_and_adopt_manual_installed_applications/brew_cask_and_adopt_manual_installed_applications.py [hunk 3]
```diff
@@ -127,85 +195,107 @@ def is_default_apple_app(app_name):
...
 
 if __name__ == "__main__":
-    INSTALL_DIR = "/Applications"
-    if len(sys.argv) > 1:
-        INSTALL_DIR = sys.argv[1]
-    main(INSTALL_DIR)
+    parser = argparse.ArgumentParser(
+        description="Adopt manually installed applications to Homebrew Cask."
+    )
+    parser.add_argument(
```
Comment (xFuture603) [inline brew_cask_and_adopt_manual_installed_applications/brew_cask_and_adopt_manual_installed_applications.py:286]: Could you also update the README.md with the new parameters please?

**Review by xFuture603 — CHANGES_REQUESTED** [review]
Thank you for your work and for your pull request!

This is indeed a great upgrade for this script. I've left a few comments as a review on this pull request.

One more thing that came up while testing. I tried the default behavior of the script with iTerm2 in this example (which was manually installed in this case) and it gave me this error:

```
Checking for "iterm"...
Cask "iterm" not found. Here are some alternatives:
1. font-ia-writer-mono
2. iterm2
3. iterm2@beta
4. iterm2@nightly
5. itermai
6. therm
Choose a cask by number (1-6), or press Enter to skip: 2
You selected: iterm2
Trying to install "iterm2" with Homebrew...
Adopt installation failed for "iterm2":
Error: The bundle short version of /opt/homebrew/Caskroom/iterm2/3.5.2/iTerm.app is 3.5.2 but is 3.4.23 for /Applications/iTerm.app!
Error: It seems the existing App is different from the one being installed.
```

I think it would be nice if it tries to uninstall and reinstall the program to avoid these errors. What do you think?

**Comment by Nemental** [comment]
@xFuture603
> I think it would be nice if it tries to uninstall and reinstall the program to avoid these errors. What do you think?

I've already thought about this, let me take a look 👍 It sounds like a good use case for a `-u`/`--upgrade` argument. But I would leave the default behavior as it is now to avoid application conflicts?!

**Comment by Nemental** [comment]
I didn't choose the way with an extra argument, instead it asks each time if the version is different from the installed one.

**Comment by xFuture603** [comment]
> he version is

Looking good now, after I've tested it again. I will merge this pull request now. Thank you!

**Review by xFuture603 — APPROVED** [review]

**Pull request merged**